t/rose-task-run/45-app-fcm-make-compat_mode.t

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------
+# Copyright (C) British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Compatibility Mode Test
+# Ensure that when Cylc Rose gets workflow config that
+# compatibility run mode isn't changed.
+# https://github.com/cylc/cylc-rose/issues/319
+#-------------------------------------------------------------------------------
+. "$(dirname "$0")/test_header"
+
+if ! fcm help make 1>/dev/null 2>&1; then
+    skip_all '"fcm make" unavailable'
+fi
+if ! gfortran --version 1>/dev/null 2>&1; then
+    skip_all '"gfortran" unavailable'
+fi
+JOB_HOST="$(rose config --default= 't' 'job-host')"
+if [[ -n "${JOB_HOST}" ]]; then
+    JOB_HOST="$(rose host-select -q "${JOB_HOST}")"
+fi
+if [[ -z "${JOB_HOST}" ]]; then
+    skip_all '"[t]job-host" not defined or not available'
+fi
```

Comment:
I don't think a job host is required to test this functionality?

Reply:
I tried it without and couldn't replicate the problem...

Reply:
I managed to replicate this with the `[remote]host` logic commented out.

Reply:
Yes, but it was failing with the other branches checked out for an unrelated reason.